subsys/bluetooth/host/addr.c

```diff
@@ -53,28 +53,30 @@ int bt_addr_le_create_static(bt_addr_le_t *addr)
 
 int bt_addr_from_str(const char *str, bt_addr_t *addr)
 {
-	int i, j;
-	uint8_t tmp;
+	const size_t strlen_addr = strlen("XX:XX:XX:XX:XX:XX");
```

Comment:
Does the strlen get evaluated at build time? If not, I’d consider using sizeof instead. We might also want to make these as public defines, if those don’t exist yet.

Reply:
> Does the strlen get evaluated at build time? If not, I’d consider using sizeof instead.

It's optimized away at any optimization level other than `-O0`, but it's not guaranteed.

Is the following what you suggest?

```suggestion
	const size_t strlen_addr = ARRAY_SIZE("XX:XX:XX:XX:XX:XX") - 1;
```

I prefer the following, keeping it simpler and just writing out the constant. The code has some tests now, so an error here should be caught.

```suggestion
	const size_t strlen_addr = 17; /* strlen("XX:XX:XX:XX:XX:XX") */
```

> We might also want to make these as public defines, if those don’t exist yet.

That could be nice, but then we have to decide what to name it :P. We already have `#define BT_ADDR_STR_LEN 18` in `addr.h`, but that counts the null-terminator.

Reply:
> That could be nice, but then we have to decide what to name it :P. We already have #define BT_ADDR_STR_LEN 18 in addr.h, but that counts the null-terminator.

It would be trivial to do 
```c
/* Subtract 1 as we don't parse the NULL terminator in this function */
const size_t strlen_addr = BT_ADDR_STR_LEN - 1;
```
though

Reply:
I got away from the problem here by replacing `strlen_addr = SOME_KNOWN_VALUE` with `len = strlen(str)`.